src/python/espressomd/visualization_opengl.pyx

```diff
@@ -783,19 +789,47 @@ class openGLLive:
         # Collect shapes and interaction type (for coloring) from constraints
         primitive_shapes = [
             'Shapes::Wall', 'Shapes::Cylinder', 'Shapes::Ellipsoid',
-            'Shapes::SimplePore', 'Shapes::Slitpore', 'Shapes::Sphere',
-            'Shapes::SpheroCylinder']
+            'Shapes::HollowConicalFrustum', 'Shapes::SimplePore',
+            'Shapes::Slitpore', 'Shapes::Sphere', 'Shapes::SpheroCylinder']
+
+        def get_shapes_from_union(union):
+            """Helper function to (recursively) get all shapes from a union.
+
+            """
+            shapes = []
+            for shape in union:
+                if shape.name() == 'Shapes::Union':
+                    shapes.extend(get_shapes_from_union(shape))
+                else:
+                    shapes.append(shape)
+            return shapes
+
+        def update_shapes_collection(collection, shape, ptype):
+            """Helper function to create a collection of shape objects.
+
+            """
+            name = shape.name()
+            if name in primitive_shapes:
+                collection[name].append([shape, ptype])
+                # visualization of HollowConicalFrustum requires gleSpiral!
+                if name == 'Shapes::HollowConicalFrustum' \
+                        and not bool(OpenGL.GLE.gleSpiral):
+                    collection['Shapes::Misc'].append([shape, ptype])
+            else:
+                coll_shape_obj['Shapes::Misc'].append([shape, ptype])
 
+        # actually create shape objects collection using the helper functions
         coll_shape_obj = collections.defaultdict(list)
-        for c in self.system.constraints:
-            if type(c) == espressomd.constraints.ShapeBasedConstraint:
-                t = c.get_parameter('particle_type')
-                s = c.get_parameter('shape')
-                n = s.name()
-                if n in primitive_shapes:
-                    coll_shape_obj[n].append([s, t])
+        for constraint in self.system.constraints:
+            if type(constraint) == espressomd.constraints.ShapeBasedConstraint:
+                ptype = constraint.get_parameter('particle_type')
```

Comment:
pep8  says: ```particle_type``` or ```p_type``` or ```part_type```

Reply:
As discussed offline, pep8 will be part of a general refactor of the visualizer.